### PR TITLE
Add basic incoming webhook documentation

### PR DESCRIPTION
**Pull request by jwilander**
You can see the page at http://mattermost-developer-documentation.s3-website-us-east-1.amazonaws.com/branches/incoming-webhooks/integrate/incoming-webhooks/

I removed the draft setting meaning this page will get built but there's no way to find it without knowing the URL. That way we can still see it on the site but no one will be able to find the integrate section until it's fully ready to go

### CODE REVIEW

**Comment by jasonblais** [comment]
I'm gonna review it in detail later. Should we redirect the docs.mattermost.com to the one at devs site?

**Comment by jwilander** [comment]
It might be good to have a link but we're still going to need documentation
on webhooks directed at admins since this is focused on developers

**Comment by jwilander** [comment]
Let's chat about it more during our 1-1 but the vision I have is that we have the admin integration guide on docs.mattermost.com that has a lot of the configuration and testing information for each integration type. Essentially it should have all the information an admin needs for using, setting up and maintaining an existing integration. It should then link to the developer docs for anyone looking to build an integration, and that's where all the development related integration information is

File: site/content/integrate/incoming-webhooks/_index.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+All webhook posts will display a `BOT` indicator next to the username in Mattermost clients, to help prevent against [phishing attacks](https://en.wikipedia.org/wiki/Phishing).
+
+### Parameters
+
+Incoming webhooks support more than just the `text` field. Here is a full list of supported parameters.
+
+| Parameter | Description | Required |
+|---|---|---|
+| text | [Markdown-formatted](https://docs.mattermost.com/help/messaging/formatting-text.html) message to display in the post | If `attachments` is not set, yes |
+| channel | Overrides the channel the message posts in.<br> Use an "@" followed by a username to send to a direct message.<br> Defaults to the channel set during webhook creation. | No |
```
Comment (lieut-data) [inline site/content/integrate/incoming-webhooks/_index.md]: As a developer, I'd want to know: can I post to any private channel if I know it by name?

File: site/content/integrate/incoming-webhooks/_index.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ok
+```
+
+All webhook posts will display a `BOT` indicator next to the username in Mattermost clients, to help prevent against [phishing attacks](https://en.wikipedia.org/wiki/Phishing).
+
+### Parameters
+
+Incoming webhooks support more than just the `text` field. Here is a full list of supported parameters.
+
+| Parameter | Description | Required |
```
Comment (lieut-data) [inline site/content/integrate/incoming-webhooks/_index.md]: Does this duplicate https://api.mattermost.com/#tag/webhooks? Should we focus on the /how/ with examples here vs. duplicating documentation?

File: site/content/integrate/incoming-webhooks/_index.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+POST /hooks/xxx-generatedkey-xxx HTTP/1.1
+Host: your-mattermost-server
+Content-Type: application/json
+Content-Length: 63
+
+{
+  "channel": "town-square",
+  "username": "test-automation",
+  "icon_url": "https://www.mattermost.org/wp-content/uploads/2016/04/icon.png",
+  "text": "#### Test results for July 27th, 2017\n<!channel> please review failed tests.\n
```
Comment (lieut-data) [inline site/content/integrate/incoming-webhooks/_index.md]: Where can I learn more about the `<!channel>` syntax?

File: site/content/integrate/incoming-webhooks/_index.md (hunk 1 above)
Comment (jwilander) [reply site/content/integrate/incoming-webhooks/_index.md]: Yeah, good point. I'll add a note for that

File: site/content/integrate/incoming-webhooks/_index.md (hunk 2 above)
Comment (jwilander) [reply site/content/integrate/incoming-webhooks/_index.md]: Nope, that endpoint is for creating a webhook and not about using one. There are some similarities but they are different

File: site/content/integrate/incoming-webhooks/_index.md (hunk 3 above)
Comment (jwilander) [reply site/content/integrate/incoming-webhooks/_index.md]: Hmm that's for slack compatibility, we should probably use the Mattermost syntax. I'll update it

Probably should link to slack compatibility info as well somewhere

File: site/content/integrate/incoming-webhooks/_index.md (hunk 2 above)
Comment (lieut-data) [reply site/content/integrate/incoming-webhooks/_index.md]: Right, of course! 👍

**Comment by jwilander** [comment]
@lieut-data updates made

**Comment by jasonblais** [comment]
Overall well done. No major feedback other than +1 :) 

1) More of a general question for dev site, but do we want to offer an easy way for community to share their integrations? E.g. having a text like https://docs.mattermost.com/developer/webhooks-incoming.html#share-your-integration
2) Would troubleshooting steps be helpful for devs? https://docs.mattermost.com/developer/webhooks-incoming.html#troubleshooting
3) Do we need to make the `__Treat this endpoint as a secret.__ Anyone who has it will be able to post messages to your Mattermost instance.` text more visible? I missed when I first scanned it. 
4) In the parameters table, are rows ordered roughly by how likely it is for someone to use the parameter?

**Comment by jasonblais** [comment]
I guess 1) might be covered by http://mattermost-developer-documentation.s3-website-us-east-1.amazonaws.com/branches/integrate/integrate/getting-started/

**Comment by jwilander** [comment]
1. Yeah it's at least partially covered by my other PR but maybe we can more links to it throughout, maybe as part of a page footer or something?
2. To an extent yes, though I think the cases covered in your link wouldn't need mentioning if we just provided better error messages in the first place (and we don't need the windows curl one since we shouldn't be teaching devs how to use curl, way beyond our scope)
3. Yeah, maybe. Not sure how to do that in a nice way though, maybe a good question for Mike or Asaad
4. Roughly, yeah. 'text', 'channel', 'username' and 'icon_url' are definitely the most likely to be used

**Pull request merged**